### PR TITLE
Fix // newline handling in freetype font

### LINKED ISSUE #2043: [Bug] Cryllic languages potentially not supporting "//" in strings
#### Describe the issue
It seems that the newline command `//` is not properly recognised in Czech, potentially applying to all Cryllic languages.

image

Code:
```lua
 pharmacy = {
    [1] = "Lékárna//",
    [2] = "Pacienti, kteří byly vyšetřeni a jejichž léčba je lék musí navštívit Lékárnu, aby si ho vzali. Jak více a více léků pro léčbu je objeveno a stanou se dostupnými, tato místnost bude více zaneprázdněná. Možná byste mohli chtít později postavit další Lékárnu.//",
    [3] = "Lékárna vyžaduje Sestru. ",
  },
```

#### Steps to Reproduce
1. Load game in Czech (Čeština)

2. Start a new campaign with defaults

3.  Build a room (I used pharmacy)

#### Expected Behavior
Recognise `//` as a newline command.

#### System Information
**CorsixTH Version:** 0.65.1 (likely previously versions)

**Operating System:** Windows 10

**Theme Hospital Version:** *CD, GOG.com, Origin, or Demo*

### Additional Info
**Workaround: Use `\n` for now.**
Ping @jansakos

### PR DESCRIPTION
**Pull request by TheCycoONE**
Fixes #2043

### CODE REVIEW

**Review by lewri — APPROVED** [review]

**Review by lewri — COMMENTED** [review]
Your second commit isn't playing well with monospace fonts (courier -- cour.ttf). It causes drops of characters.
image

Text:https://github.com/CorsixTH/CorsixTH/blob/ff7aeb4f402699620f66579cd64a22b8858b405f/CorsixTH/Lua/languages/czech.lua#L3195-L3202

**Comment by TheCycoONE** [comment]
ew

**Comment by TheCycoONE** [comment]
Fixed off by one issue with whitespace removal

File: CorsixTH/Src/th_strings.h [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+size_t discard_leading_set_bits(uint8_t& byte) {
+  size_t count = 0;
+  while ((byte & 0x80) != 0) {
+    count++;
+    byte = static_cast<uint8_t>(byte << 1);
+  }
+  byte = byte >> count;
+  return count;
+}
```
Comment (Alberth289346) [inline CorsixTH/Src/th_strings.h:20]:
```suggestion
size_t discard_leading_set_bits(uint8_t& byte) {
  size_t count = 0;
  for (uint8_t bit_val = 0x80; (byte & bit_val) != 0; bit_val >>= 1) {
    byte &= ~bit_val;
    count++;
  }
  return count;
}
```

**Comment by Alberth289346** [comment]
Some variables have a type prefix, like `sString` and `iCodepoint` and some variables don't.
Not sure you'd want to fix that now.

**Review by lewri — APPROVED** [review]

**Comment by TheCycoONE** [comment]
No, all those functions were copied over as is. I'll leave rewrite and tests on those for a future PR. The only thing I would like verified before this is merged is CJK translations - they interact with line breaks a bit differently.

**Pull request merged**